cookbook/case_studies/ml_training/spark_horovod/keras_spark_rossmann_estimator.py

```diff
@@ -323,16 +356,22 @@ def fn(v):
         df = df.withColumn(col, lookup(mapping)(df[col]))
     return df
 
-
+# %%
+# Data Preparation
+# ================
+#
+# The ``data_preparation`` function consolidates all the aforementioned data processing functions.
 def data_preparation(
     data_dir: FlyteDirectory, hp: Hyperparameters
 ) -> Tuple[float, Dict[str, List[Any]], pyspark.sql.DataFrame, pyspark.sql.DataFrame]:
     print("================")
     print("Data preparation")
     print("================")
 
+    # 'current_context' gives the handle of specific parameters in ``data_preparation`` task.
```

Comment:
Let's not have a period at the end of a comment. That's how all the comments are written.